### DISCUSSION TITLE
Custom speed to score multiplier formula

### DISCUSSION THREAD

**Discussion by WindowsMeosu** [ppy/osu#19711] [Ideas]
Issue proxied to github on behalf of [Nyaaori](https://osu.ppy.sh/u/Nyaaori)

> I would suggest using [my formula here](https://www.desmos.com/calculator/pndchcf3gz) for calculating score multiplier for custom speeds.
>
> Said function extrapolates the current multipliers of 0.3x at 75% speed and 1.12x at 150% speed out to apply for any custom speed.
>
> Essentially, my function here assumes that if one were to apply half time twice (so `0.75*0.75=0.5625`), it would result in adjusting score by another 0.3x (so `0.3*0.3=0.09`)
>
> Applying similar logic here for double time, `1.5*1.5=2.25`; `1.12*1.12=1.2544`.
>
> Adopting a formula like such would resolve or obsolete these requests/issues (duplicates included for completeness):
>
> * #7698
> * #10523
> * #12678
> * #14320
> * #19191
> * #19254
> * #19322
>
> Pseudocode version of formula, to ease implementation:
>
> ```
> if speed > 1.0:
>     multiplier = (1.12**(math.log(speed, 1.5)))
> else:
>     multiplier = (0.3**(math.log(speed, 0.75)))
> ```
>
> English version:
>
> > If input speed is greater than one,
> >
> > Take 1.12 to the power of the base 1.5 logarithm of input speed
> >
> > otherwise,
> >
> > Take 0.3 to the power of the base 0.75 logarithm of input speed
>
> ~ Nyaaori

**Comment by smoogipoo** [comment]
Sounds fine to me. I'm sure @peppy will have reservations about adding a log() in, but I don't see another way.

**Reply by peppy** [reply]
I'm fine with it, but would want to write this up in a spreadsheet/graph format and show it to a larger portion of the community to get some more feedback. And I don't see any huge hurry to make this change right now; we have enough moving pieces and this can most definitely happen a bit later in the process.

**Comment by frenzibyte** [comment]
A progressive formula for rate adjustment mods will be implemented in https://github.com/ppy/osu/pull/23177, but I'm not sure if it has been decided as the final version, and that all other proposals should be dismissed (cc @smoogipoo).

**Reply by smoogipoo** [reply]
This can remain open. I've kept it simple for the time being because I prefer/think people would rather have steppings than continuous sequences (display-wise).

For DT it's a pretty good approximation of the above: 
![image](https://user-images.githubusercontent.com/1329837/230870681-30442fb9-7c91-4495-8edb-26c692b70331.png)

For HT it's less so:
![image](https://user-images.githubusercontent.com/1329837/230871470-7c64c6b3-904f-45e4-b3b9-41b2daf32972.png)
